backend/src/schema/resolvers/posts.js

```diff
@@ -211,10 +225,68 @@ export default {
       })
       return emoted
     },
+    pinPost: async (_parent, params, context, _resolveInfo) => {
+      let pinnedPost
+      const { driver, user } = context
+      const session = driver.session()
+      const { id: userId } = user
+      let writeTxResultPromise = session.writeTransaction(async transaction => {
+        const deletePreviousRelationsResponse = await transaction.run(
+          `
+          MATCH ()-[previousRelations:PINNED]->(post:Post)
+          DELETE previousRelations
+          RETURN post
+        `,
+          { params },
+        )
+        return deletePreviousRelationsResponse.records.map(record => record.get('post').properties)
+      })
+      await writeTxResultPromise
+
+      writeTxResultPromise = session.writeTransaction(async transaction => {
+        const pinPostTransactionResponse = await transaction.run(
+          `
+            MATCH (user:User {id: $userId}) WHERE user.role = 'admin'
+            MATCH (post:Post {id: $params.id})
+            MERGE (user)-[:PINNED {createdAt: toString(datetime())}]->(post)
+            RETURN post
+         `,
+          { userId, params },
+        )
+        return pinPostTransactionResponse.records.map(record => record.get('post').properties)
+      })
+      try {
+        ;[pinnedPost] = await writeTxResultPromise
+      } finally {
+        session.close()
+      }
+      return pinnedPost
+    },
+    unpinPost: async (_parent, params, context, _resolveInfo) => {
+      let unpinnedPost
+      const session = context.driver.session()
+      const writeTxResultPromise = session.writeTransaction(async transaction => {
+        const unpinPostTransactionResponse = await transaction.run(
+          `
+          MATCH ()-[previousRelations:PINNED]->(post:Post {id: $params.id})
```

Comment:
Why not `User` here?
```suggestion
          MATCH (:User)-[previousRelations:PINNED]->(post:Post {id: $params.id})
```

Reply:
same as :point_up_2: